sql/sql_select.cc

```diff
@@ -5600,7 +5600,7 @@ make_join_statistics(JOIN *join, List<TABLE_LIST> &tables_list,
   */
 
   if (!multi_alloc_root(join->thd->mem_root,
-                        &stat, sizeof(JOIN_TAB)*(table_count),
+                        &join->join_tab, sizeof(JOIN_TAB)*(table_count),
```

Comment:
As discussed on Slack: the consequence of this is that we may end up with a partially-initialized join->join_tab array if we get a failure somewhere in the middle of this function.  This will cause issues in later call to JOIN::cleanup. So, please restore the original code here.

Reply:
Agree, reverted to the initial version